staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go

```diff
@@ -169,6 +169,35 @@ func testCheckResultFunc(t *testing.T, w watch.Interface, check func(actualEvent
 	}
 }
 
+func testCheckResultWithIgnoreFunc(t *testing.T, w watch.Interface, expectedEvents []watch.Event, ignore func(watch.Event) bool) {
+	checkIndex := 0
+	for {
+		select {
+		case event := <-w.ResultChan():
+			obj := event.Object
+			if co, ok := obj.(runtime.CacheableObject); ok {
+				event.Object = co.GetObject()
+			}
+			if ignore != nil && ignore(event) {
+				continue
+			}
+			if checkIndex < len(expectedEvents) {
+				expectNoDiff(t, "incorrect event", expectedEvents[checkIndex], event)
+				checkIndex++
+			} else {
+				t.Fatalf("cannot receive correct event, expect no event, but get a event: %+v", event)
+			}
+		case <-time.After(2 * time.Second):
```

Comment:
We still didn't fully address the problem of long-running tests and exceeding the timeout for single package tests.

So let's for now use something shorter - please do the same thing as in `testCheckNoMoreResults`

Reply:
I noticed that the timeout used in the `testCheckNoMoreResults` function is `100ms`, but we cannot guarantee that the normal bookmark event (not the initial bookmark event) will be received within `100ms`. It is possible that it will be received at `101ms`, `102ms`, or `103ms` ..., which could cause the subsequent `testCheckNoMoreResults` function to fail. Therefore, I still use `100ms` in the `testCheckResultWithIgnoreFunc` function to avoid package test taking too long, and I add a new function called `testCheckNoMoreResultsWithIgnoreFunc` to ignore the normal bookmark event. :)